### ISSUE TITLE
About the top-energy of elic and vae features.

### ISSUE THREAD

**Issue by seanxxx6** [LuizScarlet/StableCodec#13]
Hello, thank you very much for your open-source work on image compression.I would like to ask a question about the visualization of your paper Fig4 "Top energy channels learned from different encoders". 

I visualized the features of ELIC and VAE outputs in the maximum energy channel and obtained results that seem to be opposite to your paper. My ELIC visualization results are more similar to the visualization results of E-SD in your paper, while the visualization results of E-SD are more similar to those of ELIC. 
My visualization codes for energy calculation is as follows. 
`# x shape: C,H,w 
  C, H, W = x.shape

     # cal the energy 

    channel_energy = (x ** 2).view(C, -1).sum(dim=1)  # shape (C,)

    max_idx = torch.argmax(channel_energy).item()

    print("max channel index:", max_idx, "energy:", channel_energy[max_idx].item()) `

I normalized the features by max-min normalization for visualization.
The visualization images are shown below.

The elic visualization:

<img width="271" height="319" alt="Image" src="https://github.com/user-attachments/assets/a3787e95-0c1e-4433-a2c8-4831fbd0abaf" />

The vae visualization:

<img width="267" height="318" alt="Image" src="https://github.com/user-attachments/assets/2d082ad6-5d35-4d16-9044-54c1e3d9c908" />

 May I ask how energy is calculated in your method? Or perhaps it was a typographical error on your part at that time?

**Comment by LuizScarlet** [comment]
Hi @seanxxx6 , seems that your SD-VAE visualization here is in (H/8, W/8) and ELIC visualization is in (H/16, W/16). In our paper, we map the features of different encoders to (H/16, W/16), then perform the visualization. Would you mind take a look at these features? 
[/src/latent_codec.py Line122](https://github.com/LuizScarlet/StableCodec/blob/e7994cf5e18098ce3451a0d7a732559a0648b0c4/src/latent_codec.py#L122)

For your reference:
```python
def forward(self, latent, latent2):
        x = torch.cat((self.pre1(latent), self.pre2(latent2)), dim=1)

        select1 = self.pre1(latent)[0]
        energy = torch.norm(select1, dim=(1, 2), p=2)
        _, indices1 = torch.sort(energy, descending=True)

        select2 = self.pre2(latent2)[0]
        energy = torch.norm(select2, dim=(1, 2), p=2)
        _, indices2 = torch.sort(energy, descending=True)
```

**Comment by seanxxx6** [comment]
Thanks a lot for your help!! I get the similar results with your paper. That's right! The ELIC features shows more pixel-style vales.
